src/__tests__/lib/utils/hooks/SynapseAPI/QueryClientProviderWrapper.test.tsx

```diff
@@ -0,0 +1,34 @@
+import { mount } from 'enzyme'
+import React from 'react'
+import { useQuery } from 'react-query'
+import { resolveAllPending } from '../../../../../lib/testutils/EnzymeHelpers'
+import { withQueryClientProvider } from '../../../../../lib/utils/hooks/SynapseAPI/QueryClientProviderWrapper'
+
+const BasicComponentWithUseQuery: React.FC = () => {
+  const query = useQuery(['some-query-key'], () => Promise.resolve('Hello'))
+  return <div>{query.data}</div>
+}
+
+describe('Functionality', () => {
+  it('Successfully renders an application that uses useQuery', async () => {
+    const WrappedComponent = withQueryClientProvider(BasicComponentWithUseQuery)
+    const wrapper = mount(<WrappedComponent />)
+    await resolveAllPending(wrapper)
+    expect(wrapper.find('div').text()).toEqual('Hello')
+  })
+
+  it('An application that uses useQuery fails to render without the wrapper', async () => {
+    // Mocking console.error not necessary for the test, we just expect it to happen and want to suppress
+    jest.spyOn(console, 'error').mockImplementation(() => {})
+
+    try {
+      expect(mount(<BasicComponentWithUseQuery />)).toThrow(
+        expect.stringContaining(
+          'No QueryClient set, use QueryClientProvider to set one',
```

Comment:
Is looking for this exact substring going to make this test fragile (moving forward)?

Reply:
Probably will be more fragile than necessary. I think since we test both the error and non-error conditions, it's unnecessary to check the message.